### PR TITLE
Add test for sqlite_math_functions tag

### LINKED ISSUE #990: Trying to use maths function (from SQLite v3.35) does not work
Since [this PR](https://github.com/mattn/go-sqlite3/pull/926) the underlying version of SQLite should be >=3.35 (and it is). The PR even mentions the new maths function added.
However, when I try to use math functions (like `SELECT power(2,2)`) I get this error `no such function: power`.

Is there something more I need to do than use the latest version of the go-sqlite3 library?

### PR DESCRIPTION
**Pull request by lggruspe**
Hello, this is a continuation of #1037, which fixes #990.

## Changes

- Adds test to check if sqlite math functions are compiled in
- Update github workflow to include sqlite_math_functions tag in full test

@rittneje @nkt-dk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,6 +172,7 @@ go build --tags "icu json1 fts5 secure_delete"
 |  International Components for Unicode | sqlite_icu | This option causes the International Components for Unicode or "ICU" extension to SQLite to be added to the build |
 | Introspect PRAGMAS | sqlite_introspect | This option adds some extra PRAGMA statements. <ul><li>PRAGMA function_list</li><li>PRAGMA module_list</li><li>PRAGMA pragma_list</li></ul> |
 | JSON SQL Functions | sqlite_json | When this option is defined in the amalgamation, the JSON SQL functions are added to the build automatically |
+| Math Functions | sqlite_math_functions | This compile-time option enables built-in scalar math functions |
```
Comment (rittneje) [inline README.md:175]: Can you link to https://www.sqlite.org/lang_mathfunc.html? That way it's easy for people to see what all the math functions are.

File: sqlite3_opt_math_functions_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	defer db.Close()
+
+	queries := []string{
+		`SELECT acos(1)`,
+		`SELECT log(10, 100)`,
+		`SELECT power(2, 2)`,
+	}
+
+	for _, query := range queries {
+		if _, err := db.Exec(query); err != nil {
```
Comment (rittneje) [inline sqlite3_opt_math_functions_test.go:24]: You should use `db.QueryRow` for `SELECT`, not `db.Exec`.

File: sqlite3_opt_math_functions.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2019 Yasuhiro Matsumoto <mattn.jp@gmail.com>.
```
Comment (rittneje) [inline sqlite3_opt_math_functions.go:1]: @mattn What's the expectation for the copyright statement on new files?

File: README.md (hunk 1 above)
Comment (lggruspe) [reply README.md]: Done

File: sqlite3_opt_math_functions_test.go (hunk 2 above)
Comment (lggruspe) [reply sqlite3_opt_math_functions_test.go:24]: I also considered `QueryRow`, but it defers the error to `Scan`, and the result isn't really needed for the test. Maybe `Query` would be a better idea?

File: sqlite3_opt_math_functions.go (hunk 3 above)
Comment (mattn) [reply sqlite3_opt_math_functions.go:1]: s/2019/2022/

File: sqlite3_opt_math_functions.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// Copyright (C) 2018 G.J.R. Timmer <gjr.timmer@gmail.com>.
```
[the remark this answers is not in the document]
Comment (lggruspe) [reply sqlite3_opt_math_functions.go]: done

File: sqlite3_opt_math_functions_test.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	defer db.Close()
+
+	queries := []string{
+		`SELECT acos(1)`,
+		`SELECT log(10, 100)`,
+		`SELECT power(2, 2)`,
+	}
+
+	for _, query := range queries {
+		if _, err := db.Query(query); err != nil {
```
Comment (rittneje) [inline sqlite3_opt_math_functions_test.go:24]: Right now you are dropping the `Rows` object. That means you are only checking for an error from `sqlite3_prepare_v2` and not `sqlite3_step`. For completeness we should check both, which is why `QueryRow` would be preferable.

```go
var result float64
if err := db.QueryRow(query).Scan(&result); err != nil {
    t.Errorf("invoking math function query %q: %v", query, err)
}
// For the purposes of this test, we don't care about the output, just that the query succeeded.
```

Comment (lggruspe) [reply sqlite3_opt_math_functions_test.go]: I see... thanks for the review :)

**Comment by mattn** [comment]
Seems okay to me.

**Review by mattn — APPROVED** [review]

**Comment by rittneje** [comment]
Failure is due to unrelated issue introduced in #1085.

**Pull request merged**